code/modules/SCPmodular/scps/euclid/SCP-173.dm

```diff
@@ -0,0 +1,212 @@
+				if(!M.eye_blind)
+					if(next_blinks[M] == null)
+						next_blinks[M] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					if(next_blinks[M.occupant] == null)
+						next_blinks[M.occupant] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M.occupant
+	return null
+
+
+
+
+
+
+/*    VIEWLOOP
+	// This loop will, at most, loop twice.
+	for(var/atom/check in check_list)
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					return M.occupant
+	return null
+*/
+
+
+/mob/living/simple_animal/hostile/scp_173/Move(turf/NewLoc)
+	if(can_be_seen(NewLoc))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot move, there are eyes on you!</span>")
+		return 0
+	return ..()
+
+/mob/living/simple_animal/hostile/scp_173/movement_delay()
+	return -5
+
+
+
+/mob/living/simple_animal/hostile/scp_173/Life()
+	. = ..()
+	if (isobj(loc))
+		return
+	var/list/our_view = view(src, 23)
+	for(var/A in next_blinks)
+		if(!(A in our_view))
+			next_blinks[A] = null
+			continue
+		if(world.time >= next_blinks[A])
+			var/mob/living/carbon/human/H = A
+			if(H.stat) // Sleeping or dead people can't blink!
+				next_blinks[A] = null
+				continue
+			H.visible_message("<span class='notice'>[H] blinks.</span>")
+			H.blind_eyes(2)
+			next_blinks[H] = 10+world.time+rand(15 SECONDS, 60 SECONDS)
+
+
+
+
+
+/mob/living/simple_animal/hostile/scp_173/sentience_act()
+	faction -= "neutral"
+
+
+/mob/living/simple_animal/hostile/scp_173/AttackingTarget(/*var/atom/A*/) // doesnt work?
+	if(can_be_seen(get_turf(loc)))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot attack, there are eyes on you!</span>")
+		return FALSE
+	else
+		var/mob/living/H = target
+		if(isliving(target))
+			if(target == src)
+				to_chat(src, "<span class='warning'><I>You can't hit yourself!</I></span>")
+			if(ishuman(H))
+				visible_message("<span class='danger'>[src] snaps [H]'s neck!</span>")
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+			if(!ishuman(H))
+				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+			if(!isliving(target))
+				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
```

Comment:
All of this code is the same except for one of the visible messages. You also already have an isliving check.

Reply:
`		var/mob/living/H = A
		if(isliving(A))
			if(A == src)
				to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking ourselves?</I></span>")
				return
			else 
			visible_message(pick("<span class='danger'>[src] snaps [H]'s neck!</span>" , "<span class='danger'>[src] crushes [H] with raw force!</span>"))
				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
				H.death()
			else
				return // this is just for error catching`
Possible fix? untested yet.